operator/cmd/mesh/operator-remove.go

```diff
@@ -78,13 +78,17 @@ func operatorRemove(cmd *cobra.Command, args *RootArgs, orArgs *operatorRemoveAr
 		l.LogAndFatal(err)
 	}
 
+	switch {
 	// If the user is performing purge but also specified a revision, an error message
 	// should be displayed and do nothing
-	if orArgs.purge && orArgs.revision != "" {
+	case orArgs.purge && orArgs.revision != "":
 		orArgs.revision = ""
-		l.LogAndFatal("At most one of the --revision or --purge flags could be set\n")
-	} else if orArgs.revision == "default" {
+		l.LogAndPrint("Purge remove will remove all Istio operator controller, ignoring the specified revision\n")
+	case orArgs.purge:
 		orArgs.revision = ""
+	case orArgs.revision == "":
```

Comment:
I think we never set empty revision string to `default` in the code logic.

Reply:
I've checked the old version, the `operator remove` behavior was if you were specifying a revision, the revision would be deleted, and if you were not specifying a revision, all revisions would be deleted.

I think `case orArgs.purge` is correct since it's gonna get all of the revisions, and do the deletion. However, I don't think `case orArgs.revision == ""` is reasonable. The better approach should be to just use the same logic as what `uninstall` does, if you are not specifying revision, it will throw an error `Error: at least one of the --revision (or --set revision=<revision>), --filename or --purge flags must be set`, then if we want to remove the default operator controller, we must do something like `operator remove -r default`, and you won't have this kind of post-processing logics.

Reply:
Now these modifications are adjustments made without changing usage, and this logic was decided in this discussion: https://github.com/istio/istio/pull/41548#discussion_r1001892926

Reply:
I followed the suggestion and updated the code, please review again, thanks